### PR TITLE
test(ui5-toolbar): async silently failing tests rewritten

### PR DESCRIPTION
**Pull request by PetyaMarkovaBogdanova**

### CODE REVIEW

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 1]
```diff
@@ -360,50 +360,84 @@ describe("Toolbar Select", () => {
...
+            <ToolbarButton
+                icon="sap-icon://add"
+                accessible-name="Add"
+                accessible-name-ref="btn"
+            ></ToolbarButton>
+        </Toolbar>
+		).then(() =>	{
+			const accButton = document.querySelector(
+				"ui5-toolbar-button[accessible-name]"
+			) as ToolbarButton;
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: you can pass these objects directly into the JSX template

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 2]
```diff
@@ -360,50 +360,84 @@ describe("Toolbar Select", () => {
...
+	})
+
+
+	it("Should render the button with the correct text inside the popover", () => {
+		cy.viewport(100, 1080);
+
+		cy.get("#otb_d")
+			.shadow()
+			.find(".ui5-tb-overflow-btn")
+			.click();
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: realClick instead of click

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 3]
```diff
@@ -360,50 +360,84 @@ describe("Toolbar Select", () => {
...
+				hasPopup: "dialog",
+			};
+		})
+	})
+
+
+	it("Should render the button with the correct text inside the popover", () => {
+		cy.viewport(100, 1080);
+
+		cy.get("#otb_d")
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: select by [custom-tag] not by id

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 4]
```diff
@@ -360,50 +360,84 @@ describe("Toolbar Select", () => {
...
+			.shadow()
+			.find(".ui5-tb-overflow-btn")
+			.click();
+
+		cy.get("ui5-toolbar-button").first().shadow().find("ui5-button")
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: select by attrbiute, not by tag name

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 5]
```diff
@@ -360,50 +360,76 @@ describe("Toolbar Select", () => {
...
-						expect($button).to.have.attr("tooltip", "Add");
-					});
-				});
-			});
-		});
+//ToolbarButton
+describe("ToolbarButton", () => {
+	beforeEach(() => {
+
+		cy.mount(
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: id is obsolete

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 6]
```diff
@@ -360,50 +360,76 @@ describe("Toolbar Select", () => {
...
+			<Toolbar id="otb_d">
+            <ToolbarButton
+                text="Back"
+                design="Emphasized"
+                disabled
+                icon="sap-icon://add"
+                end-icon="sap-icon://employee"
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: use endIcon instead end-icon when setting props

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 7]
```diff
@@ -360,50 +360,76 @@ describe("Toolbar Select", () => {
...
-					expect($button).to.have.attr("accessible-name-ref", "btn");
-				});
-			});
-		});
+		cy.get("[ui5-toolbar]")
+			.shadow()
+			.find(".ui5-tb-overflow-btn")
+			.realClick();
+
+			cy.get("ui5-toolbar-button[accessible-name]").shadow().find(".ui5-tb-button")
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx]: by [custom-tag] attribute, not by tag name

**Review by nnaydenow — COMMENTED** [review]
There is one more test: it("Should move button with alwaysOverflow priority to overflow popover", async () =>

File: packages/main/cypress/specs/Toolbar.cy.tsx [hunk 8]
```diff
@@ -175,39 +175,40 @@ describe("Toolbar general interaction", () => {
...
 		cy.wait(500);
 
-		const otb = cy.get("#otb_d");
-
-		cy.get("otb")
+		// Select the toolbar by tag name
+		cy.get("ui5-toolbar")
 			.shadow()
 			.find(".ui5-tb-overflow-btn")
 			.click();
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Toolbar.cy.tsx:193]: realclick

**Comment by PetyaMarkovaBogdanova** [comment]
Will be covered in https://github.com/SAP/ui5-webcomponents/pull/11972 and https://github.com/SAP/ui5-webcomponents/pull/11861
